subprojects/common/http-client/src/main/kotlin/com/avito/http/HttpClientProvider.kt

```diff
@@ -0,0 +1,25 @@
+package com.avito.http
+
+import com.avito.android.stats.StatsDSender
+import com.avito.http.internal.StatsHttpEventListener
+import com.avito.time.TimeProvider
+import okhttp3.OkHttpClient
+
+public class HttpClientProvider(
+    private val statsDSender: StatsDSender,
+    private val timeProvider: TimeProvider
+) {
+
+    private val builder = OkHttpClient.Builder()
+
+    public fun provide(): OkHttpClient.Builder {
+        return builder.eventListenerFactory {
+            StatsHttpEventListener(
```

Comment:
We propagate all dependencies for StatsHttpEventListener in HttpClientProvider.
Let's use DI principle and pass EventListener to HttpClientProvider.

Reply:
on second though it doesn't make much sense now
`HttpClientProvider` is exactly the place we want to hide concrete listener creation

Reply:
I don't think so, but I don't want to spend time arguing so do what you should